### PR TITLE
Added support for deploy_nft to deploy an ERC721 contract

### PR DESCRIPTION
**Pull request by rohan-agarwal-coinbase**
### What changed? Why?
- Added support for deploy_nft to deploy an ERC721 contract

### Testing
- Unit tests pass
- Did an e2e test and was able to deploy the contract successfully followed by invoke_contract to mint a few NFTs!

### CODE REVIEW

File: lib/coinbase/smart_contract.rb [hunk 1]
```diff
@@ -72,6 +72,31 @@ def self.create_token_contract(
       new(contract)
     end
 
+    def self.create_nft_contract(
```
Comment (John-peterson-coinbase) [inline lib/coinbase/smart_contract.rb:75]: please document this method

File: lib/coinbase/address/wallet_address.rb
Comment (John-peterson-coinbase) [inline lib/coinbase/address/wallet_address.rb:1]: please add tests for `wallet_address.rb` changes to `wallet_address_spec.rb`

**Review by John-peterson-coinbase — COMMENTED** [review]
LGTM! 2 comments about docs and tests.

File: lib/coinbase/address/wallet_address.rb
Comment (rohan-agarwal-coinbase) [reply lib/coinbase/address/wallet_address.rb:1]: Done, also added for `deploy_token` too

**Review by John-peterson-coinbase — APPROVED** [review]

**Pull request merged**